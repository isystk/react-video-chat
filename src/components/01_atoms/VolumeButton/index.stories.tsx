import { Meta, Story } from '@storybook/react'
import React from 'react'
import VolumeButton from './index'
import MainService from '@/services/main'

export default {
  title: '01_atoms/VolumeButton',
  component: VolumeButton,
} as Meta

const main = {} as MainService

const Template: Story = () => <VolumeButton main={main} />

export const Primary = Template.bind({})
Primary.storyName = 'プライマリ'
