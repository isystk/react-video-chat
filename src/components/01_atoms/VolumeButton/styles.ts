import { css, keyframes } from '@emotion/css'

export const volumeButton = css``
