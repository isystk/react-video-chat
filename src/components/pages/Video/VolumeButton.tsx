import React, { MutableRefObject, FC } from 'react'

import IconButton from '@material-ui/core/IconButton'
import VolumeOffIcon from '@material-ui/icons/VolumeOff'
import VolumeUpIcon from '@material-ui/icons/VolumeUp'
import { OverridableComponent } from '@material-ui/core/OverridableComponent'
import { SvgIconTypeMap } from '@material-ui/core'
import Main from '@/services/main'

type Props = {
  isLocal: boolean
  muted: OverridableComponent<SvgIconTypeMap>
  refVolumeButton: MutableRefObject<null>
  main: Main
  setMuted: (value: ((prevState: boolean) => boolean) | boolean) => void
}

const VolumeButton: FC<Props> = ({
  isLocal,
  refVolumeButton,
  main,
  color = 'black',
}) => {
  const Icon = isLocal && main.self.muted ? VolumeOffIcon : VolumeUpIcon

  return (
    <IconButton
      aria-label="switch mute"
      onClick={async () => {
        if (isLocal) {
          // 音声のオン・オフを切り替える
          await main.toggleAudio()
        }
      }}
      ref={refVolumeButton}
    >
      <Icon style={{ fill: color }} />
    </IconButton>
  )
}

export default VolumeButton
