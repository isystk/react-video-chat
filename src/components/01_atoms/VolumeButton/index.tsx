import React, { FC } from 'react'

import IconButton from '@material-ui/core/IconButton'
import VolumeOffIcon from '@material-ui/icons/VolumeOff'
import VolumeUpIcon from '@material-ui/icons/VolumeUp'
import { ContainerProps, WithChildren } from 'types'
import { useStyles } from './styles'
import { connect } from '@/components/hoc'

/** VolumeButtonProps Props */
export type VolumeButtonProps = WithChildren & {
  main
  isLocal
  refVolumeButton
  color
}
/** Presenter Props */
export type PresenterProps = VolumeButtonProps & { classes }

/** Presenter Component */
const VolumeButtonPresenter: FC<PresenterProps> = ({
  main,
  classes,
  isLocal,
  refVolumeButton,
  color,
  Icon,
  ...props
}) => (
  <>
    <IconButton
      aria-label="switch mute"
      onClick={async () => {
        if (isLocal) {
          // 音声のオン・オフを切り替える
          alert('todo')
        }
      }}
      ref={refVolumeButton}
    >
      <Icon style={{ fill: color }} />
    </IconButton>
  </>
)

/** Container Component */
const VolumeButtonContainer: React.FC<
  ContainerProps<VolumeButtonProps, PresenterProps>
> = ({
  presenter,
  children,
  main,
  isLocal,
  refVolumeButton,
  color = 'black',
  ...props
}) => {
  const classes = useStyles()
  const Icon = isLocal && main.self.muted ? VolumeOffIcon : VolumeUpIcon
  return presenter({
    children,
    main,
    isLocal,
    refVolumeButton,
    color,
    classes,
    Icon,
    ...props,
  })
}

export default connect<VolumeButtonProps, PresenterProps>(
  'VolumeButton',
  VolumeButtonPresenter,
  VolumeButtonContainer
)
