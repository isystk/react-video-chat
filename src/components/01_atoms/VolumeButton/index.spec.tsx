import React, { useState } from 'react'
import renderer from 'react-test-renderer'
import VolumeButton from './index'
import '@testing-library/jest-dom/extend-expect'
import { renderHook } from '@testing-library/react-hooks'
import Main from '@/services/main'

describe('VolumeButton', () => {
  it('Match Snapshot', () => {
    const stateMain = renderHook(() => useState<Main | null>(null))
    const [, setAppRoot] = stateMain.result.current
    const main = new Main(setAppRoot)
    if (main === null) return
    const component = renderer.create(<VolumeButton main={main} />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
