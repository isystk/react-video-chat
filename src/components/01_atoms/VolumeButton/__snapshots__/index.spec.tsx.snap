// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VolumeButton Match Snapshot 1`] = `
<button
  aria-label="switch mute"
  className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-0 css-78trlr-MuiButtonBase-root-MuiIconButton-root"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
    data-testid="VolumeUpIcon"
    focusable="false"
    style={
      Object {
        "fill": "black",
      }
    }
    viewBox="0 0 24 24"
  >
    <path
      d="M3 9v6h4l5 5V4L7 9H3zm13.5 3c0-1.77-1.02-3.29-2.5-4.03v8.05c1.48-.73 2.5-2.25 2.5-4.02zM14 3.23v2.06c2.89.86 5 3.54 5 6.71s-2.11 5.85-5 6.71v2.06c4.01-.91 7-4.49 7-8.77s-2.99-7.86-7-8.77z"
    />
  </svg>
</button>
`;
